### PR TITLE
Printf floating point numbers

### PR DESCRIPTION
**Pull request by contrun**

### CODE REVIEW

File: include/ckb-c-stdlib/libc/src/impl.c [hunk 1]
```diff
@@ -283,62 +294,73 @@ void *memmove(void *dest, const void *src, size_t n) {
...
-#define HASZERO(x) (((x)-ONES) & ~(x) & HIGHS)
-
-void *memchr(const void *src, int c, size_t n)
-{
-	const unsigned char *s = src;
-	c = (unsigned char)c;
+#define ALIGN (sizeof(size_t) - 1)
+#define ONES ((size_t)-1 / UCHAR_MAX)
+#define HIGHS (ONES * (UCHAR_MAX / 2 + 1))
+#define HASZERO(x) (((x)-ONES) & ~(x)&HIGHS)
```
Comment (XuJiandong) [inline include/ckb-c-stdlib/libc/src/impl.c:339]: Add prefix "CKB_" to these macros.

File: include/ckb-c-stdlib/libc/src/impl.c [hunk 2]
```diff
@@ -1492,9 +1822,13 @@ int ckb_printf(const char *format, ...) { return 0; }
...
-/* We need to use inline asm for easier compilation, https://stackoverflow.com/a/42358235. */
-/* We need __attribute__((naked)) to remove prolog and epilog, https://stackoverflow.com/a/42637729 */
+/* Copied from
+ * https://github.com/bminor/musl/blob/46d1c7801bb509e1097e8fadbaf359367fa4ef0b/src/setjmp/riscv64/setjmp.S
+ */
+/* We need to use inline asm for easier compilation,
+ * https://stackoverflow.com/a/42358235. */
+/* We need __attribute__((naked)) to remove prolog and epilog,
+ * https://stackoverflow.com/a/42637729 */
 int setjmp(jmp_buf b) {
```
Comment (XuJiandong) [inline include/ckb-c-stdlib/libc/src/impl.c:1832]: In comment, it said "__attribute__((naked))". But why isn't it applied?

File: include/ckb-c-stdlib/libc/src/impl.c [hunk 3]
```diff
@@ -14,14 +14,14 @@
 
 #include <internal/atomic.h>
 #include <internal/types.h>
+#include <limits.h>
+#include <locale.h>
```
Comment (XuJiandong) [inline include/ckb-c-stdlib/libc/src/impl.c:18]: Maybe we don't need this file (locale.h)?

**Review by XuJiandong — CHANGES_REQUESTED** [review]

File: include/ckb-c-stdlib/libc/src/impl.c (hunk 3 above)
Comment (contrun) [reply include/ckb-c-stdlib/libc/src/impl.c:18]: We need it here https://github.com/contrun/ckb-lua/blob/6cec1375f1a81a0ad175fe1a4d91f0867bb269b1/include/ckb-c-stdlib/libc/src/impl.c#L623-L652 .

**Review by XuJiandong — APPROVED** [review]

**Pull request merged**